modules/miner.go

```diff
@@ -14,6 +15,15 @@ type MinerInfo struct {
 	Address        types.UnlockHash
 }
 
+// MinerWork is all the information a miner needs to do work
+type MinerWork struct {
+	Block      types.Block
+	ParentID   types.BlockID
+	Nonce      uint64
+	MerkleRoot crypto.Hash
+	Target     types.Target
+}
```

Comment:
Oh, right, sending the whole block was the hack I was using until I could figure out exactly what information needs to be sent back to the network. I'm aware that I only need those three parameters, but the host program I'm using to communicate with the FPGA needs to know the rest of the block information in order to submit it. As for the target, that needs to be sent to the FPGA so it can do the comparison internally--this saves it from having to send each hash over the wire to the host (very slow).